### ISSUE TITLE
RasterReader cannot open none-GDAL rasters

### ISSUE THREAD

**Issue by jakimowb** [EnMAP-Box/enmap-box#857]
The RasterReader fails to read QgsRasterLayers that use other data provider than GDAL (which is the case for 6 of 7 standard QGIS QgsRasterDataProviders)

The `openWithGdal` keyword also seems to be without effect.

How to reproduce:
```python
from enmapboxprocessing.rasterreader import RasterReader
from enmapbox.testing import start_app
from qgis.core import QgsRasterLayer
from enmapbox.qgispluginsupport.qps.utils import rasterArray

app = start_app()

wcsUri = r'dpiMode=7&identifier=OneGDev:WesternMed-MCol&tilePixelRatio=0&url=https://ogcdev.bgs.ac.uk/geoserver/OneGDev/wcs?acceptversions%3D2.0.1'
lyr = QgsRasterLayer(wcsUri, 'WCS Example', providerType='wcs')

assert lyr.isValid()
data = rasterArray(lyr)
print(data.shape)
s = ""

# now we would expect the RasterReader reads without GDAL
reader = RasterReader(lyr, openWithGdal=False)
```

relates to #764

**Comment by janzandr** [comment]
Hi @jakimowb, currently, the **RasterReader** only supports GDAL raster.

If we want to, we can also support **Web Coverage Service** (WCS). So far, there was no need for it.

**Comment by jakimowb** [comment]
It's not only because of WCS. There are several other data providers, e.g. for virtual rasters. Also the spectral processing widget uses its own rasterdataprovider to avoid data redundancy.

**Comment by jakimowb** [comment]
If the RasterReader supports GDAL sources only, it should be better renamed.

However, I think it would be better if it's features present a true superset of features of a standard QgsRasterLayer. It should work with any rasterlayer instance, and provides additional options in case of GDAL rasterproviders.

**Comment by janzandr** [comment]
Yes, totally agree, but let's have a seperate Issue for each new provider, which we need/want to support.

**Comment by jakimowb** [comment]
I don't think it is necessary to create an issue for each single provider. The RasterReader just should work fine with any QgsRasterData provider. On top, it might add value in case of providers like gdal, e.g. by reading metadata from extra files. However, it should still work generically with any QgsRasterData interface.

**Comment by janzandr** [comment]
WMS provider should work now (see #865)

**Comment by janzandr** [comment]
@jakimowb, which provider should be implemented next? Anything urgent before the v3.14 release?

**Comment by jakimowb** [comment]
The RasterReader should generally accept all QgsRasterLayers/QgsRasterDataProviders.

I'll enhance (somewhen) the enmapbox.testing.TestObject to provide a set of rasterlayers with different providers. See #870

**Comment by janzandr** [comment]
Waiting for enmapbox.testing.TestObject enhancement.
